### PR TITLE
Story3 task2

### PR DESCRIPTION
**Pull request by robin-hodge**
All changes added

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -20,31 +27,50 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
     })
     stringForHtml += '</p>'
     textToTypeContainer.innerHTML += stringForHtml
+    let upcomingWord = document.getElementById(0)
+    upcomingWord.style.color = ("orange")
+    upcomingWordLength = document.getElementById(0).clientWidth
```
Comment (melody-sylvestre) [inline script.js]: you could replace `document.getElementById(0)` by `upcomingWord`

File: script.js [hunk 2]
```diff
@@ -20,31 +27,50 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+        } else {
             textInput.value = ''
+            wordJustFinished.style.color = ("red")
         }
-    }
-})
 
+        let upcomingWord = document.getElementById(wordIndexCount + 1)
+        
+        upcomingWordLength = document.getElementById(wordIndexCount + 1).clientWidth //again, not sure how well named this is
```
Comment (melody-sylvestre) [inline script.js]: we think this name makes sense - you can remove this comment :)

File: script.js [hunk 3]
```diff
@@ -20,31 +27,50 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+        totalLengthOfWords += upcomingWordLength
 
+        if (totalLengthOfWords >= 864.1) {
+            
+            totalLengthOfWords = 0
+            totalLengthOfWords += upcomingWordLength
+            document.getElementById("textToTypeContainer").scroll({
```
Comment (melody-sylvestre) [inline script.js]: you should re-use the variable textToTypeContainer (declared line 1)

File: script.js [hunk 4]
```diff
@@ -20,31 +27,50 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+                top: scrollPixels += 90.75,
```
Comment (melody-sylvestre) [inline script.js]: Lines 65 and 60 - could you add comments to explain why you chose those values (.e.g `//this is the width of the box in pixels`)

Comment (robin-hodge) [reply script.js]: Done - more responsive too!

File: script.js (hunk 1 above)
Comment (robin-hodge) [reply script.js]: Done!

**Comment by melody-sylvestre** [comment]
there is a contrast error with Wave on orange words

**Comment by melody-sylvestre** [comment]
All good for me, well done guys :) !

**Review by melody-sylvestre — APPROVED** [review]

**Pull request merged**